kolena/_experimental/object_detection/dataset.py

```diff
@@ -0,0 +1,246 @@
+# Copyright 2021-2024 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import itertools
+from collections import defaultdict
+from typing import Any
+from typing import cast
+from typing import Dict
+from typing import List
+from typing import Literal
+from typing import Union
+
+import pandas as pd
+
+from kolena import dataset
+from kolena._experimental.object_detection.utils import compute_optimal_f1_threshold
+from kolena._experimental.object_detection.utils import compute_optimal_f1_threshold_multiclass
+from kolena._experimental.object_detection.utils import filter_inferences
+from kolena.annotation import LabeledBoundingBox
+from kolena.annotation import ScoredLabel
+from kolena.annotation import ScoredLabeledBoundingBox
+from kolena.errors import IncorrectUsageError
+from kolena.metrics import InferenceMatches
+from kolena.metrics import match_inferences
+from kolena.workflow.metrics import match_inferences_multiclass
+from kolena.workflow.metrics import MulticlassInferenceMatches
+
+
+def single_class_datapoint_metrics(bbox_matches: InferenceMatches, thresholds: float) -> Dict[str, Any]:
+    tp = [inf for _, inf in bbox_matches.matched if inf.score >= thresholds]
+    fp = [inf for inf in bbox_matches.unmatched_inf if inf.score >= thresholds]
+    fn = bbox_matches.unmatched_gt + [gt for gt, inf in bbox_matches.matched if inf.score < thresholds]
+    scores = [inf.score for inf in tp + fp]
+    return dict(
+        TP=tp,
+        FP=fp,
+        FN=fn,
+        matched_inference=[inf for _, inf in bbox_matches.matched],
+        unmatched_ground_truth=[gt for gt in bbox_matches.unmatched_gt],
```

Comment:
nit: seems like don't need to use list comprehension b/c bbox_matches.unmatched_gt is a list of gt already